x-pack/plugins/security_solution/server/endpoint/endpoint_app_context_services.ts

```diff
@@ -104,92 +105,128 @@ export class EndpointAppContextService {
+        this.setupDependencies.securitySolutionRequestContextFactory,
         alerting,
         licenseService,
         exceptionListsClient,
+        this.setupDependencies.cloud,
+        productFeaturesService
+      )
+    );
+
+    registerFleetCallback(
+      'packagePolicyPostCreate',
+      getPackagePolicyPostCreateCallback(logger, exceptionListsClient)
+    );
+
+    registerFleetCallback(
+      'packagePolicyUpdate',
+      getPackagePolicyUpdateCallback(
+        logger,
+        licenseService,
         featureUsageService,
         endpointMetadataService,
+        this.setupDependencies.cloud,
         esClient,
-        productFeaturesService,
-        savedObjectsClient,
-      } = dependencies;
-
-      registerIngestCallback(
-        'agentPolicyCreate',
-        getAgentPolicyCreateCallback(logger, productFeaturesService)
-      );
-      registerIngestCallback(
-        'agentPolicyUpdate',
-        getAgentPolicyUpdateCallback(logger, productFeaturesService)
-      );
-
-      registerIngestCallback(
-        'packagePolicyCreate',
-        getPackagePolicyCreateCallback(
-          logger,
-          manifestManager,
-          this.setupDependencies.securitySolutionRequestContextFactory,
-          alerting,
-          licenseService,
-          exceptionListsClient,
-          this.setupDependencies.cloud,
-          productFeaturesService
-        )
-      );
-
-      registerIngestCallback(
-        'packagePolicyPostCreate',
-        getPackagePolicyPostCreateCallback(logger, exceptionListsClient)
-      );
+        productFeaturesService
+      )
+    );
 
-      registerIngestCallback(
-        'packagePolicyUpdate',
-        getPackagePolicyUpdateCallback(
-          logger,
-          licenseService,
-          featureUsageService,
-          endpointMetadataService,
-          this.setupDependencies.cloud,
-          esClient,
-          productFeaturesService
-        )
-      );
+    registerFleetCallback(
+      'packagePolicyPostDelete',
+      getPackagePolicyDeleteCallback(exceptionListsClient, soClient)
+    );
+  }
 
-      registerIngestCallback(
-        'packagePolicyPostDelete',
-        getPackagePolicyDeleteCallback(exceptionListsClient, savedObjectsClient)
-      );
+  /**
+   * Property providing access to saved objects client factory
+   */
+  public get savedObjects(): SavedObjectsClientFactory {
+    if (!this.savedObjectsFactoryService) {
+      throw new EndpointAppContentServicesNotStartedError();
     }
 
-    if (this.startDependencies.registerListsServerExtension) {
-      const { registerListsServerExtension } = this.startDependencies;
-
-      registerListsPluginEndpointExtensionPoints(registerListsServerExtension, this);
-    }
+    return this.savedObjectsFactoryService;
   }
 
-  public stop() {}
-
   private getFleetAuthzService(): FleetStartContract['authz'] {
-    if (!this.startDependencies?.fleetAuthzService) {
+    if (!this.startDependencies?.fleetStartServices) {
```

Comment:
should we also check for `.authz` , or not necessary?

Reply:
Its not necessary